crates/pallet-domains/src/lib.rs

```diff
@@ -993,15 +971,43 @@ impl<T: Config> Pallet<T> {
             if let Some(domain_config) =
                 DomainRegistry::<T>::get(domain_id).map(|obj| obj.domain_config)
             {
-                DomainTxRangeState::<T>::mutate(domain_id, |maybe_state| {
-                    let expected_bundles_per_interval =
-                        u64::from(domain_config.target_bundles_per_block)
-                            * T::DomainTxRangeAdjustmentInterval::get();
-                    if let Some(state) = maybe_state {
-                        state.on_finalize(
-                            T::DomainTxRangeAdjustmentInterval::get(),
-                            expected_bundles_per_interval,
+                DomainTxRangeState::<T>::mutate(domain_id, |maybe_tx_range_state| {
+                    if let Some(tx_range_state) = maybe_tx_range_state {
+                        let tx_range_adjustment_interval =
+                            T::DomainTxRangeAdjustmentInterval::get();
+
+                        tx_range_state.interval_blocks += 1;
+
+                        if tx_range_state.interval_blocks < tx_range_adjustment_interval {
+                            return;
+                        }
```

Comment:
According to the [spec](https://www.notion.so/subspacelabs/Domains-v2-Specification-WIP-3fb0ec6e4d204c4881a7df50ef58da8f?pvs=4#03d4462ac2a740c8baecb0841f07f6d2), the `interval_blocks` and `tx_range_adjustment_interval` should be the domain block but now we are using the consensus block as `update_domain_tx_range` is called whenever a consensus block is finalized.

Reply:
This was using consensus block earlier as well though I agree on using the DomainBlock for this. Not a blocker and can be handled once the Block tree PR is in